torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1728,19 +1730,58 @@ struct to_ir {
   // raise a
   //
   // We ignore the expression following raise
-  void emitRaise(const SourceRange& loc) {
-    const std::string exception = "Exception";
-    auto string_input = insertConstant(*graph, exception, loc);
-    graph->insert(prim::RaiseException, {string_input}, {}, loc);
+  void emitRaise(const Raise& raise) {
+    auto sv = emitSugaredExpr(raise.expr(), 1);
+    Value* value = nullptr;
+
+    auto exception_instance = std::dynamic_pointer_cast<ExceptionMessageValue>(sv);
+    if (auto exception_instance =
+            std::dynamic_pointer_cast<ExceptionMessageValue>(sv)) {
+      // The typical case, an instance of the exception class was thrown:
+    //    raise RuntimeError("error")
+      value = exception_instance->getValue();
+    } else if (
+        auto exception_class = std::dynamic_pointer_cast<ExceptionValue>(sv)) {
+      // A bare exception was thrown so add an empty message. e.g.
+      //    raise RuntimeError
+      value = insertConstant(*graph, "", raise.range());
+    } else {
+      // The raise was not followed by an exception (i.e. it was something like
+      // `raise "error"` instead of `raise RuntimeError("error")`)
+      throw ErrorReport(raise.range())
+          << "exceptions must derive from BaseException";
+    }
+
+    // TODO: Fix hack (some things like `ValueError` can take non-string
+    // arguments) Since exceptions are still sugared values and don't have an
+    // `attr()` defined, this should be ok for now
+    if (!value->type()->isSubtypeOf(StringType::get())) {
+      value = graph->insert(aten::str, {value});
```

Comment:
This is going to fail at runtime instead of compilation...

Reply:
How would it fail at runtime? If it doesn't work there will be a schema-mismatch

Reply:
Yea i take this back, insert is the api that checks schemas